portfolio-form-submission.js

```diff
@@ -90,22 +102,51 @@ function AddWorkExpToList(event) {
     day: "numeric",
   });
 
-  jobDuration.textContent = startDate + " - " + endDate;
-  jobDescription.textContent = document.getElementById("description").value;
+  jobDuration.innerHTML = startDate + " - " + endDate;
+  jobDescription.innerHTML = document.getElementById("description").value;
 
   workExpItem.appendChild(companyName);
   workExpItem.appendChild(jobDuration);
   workExpItem.appendChild(jobDescription);
   document.querySelector(".exp-list").appendChild(workExpItem);
 
+  workExpList.push({
+    companyName: companyName.textContent,
+    duration: jobDuration.textContent,
+    description: jobDescription.textContent,
+  });
+
+  localStorage.setItem("work-exp", JSON.stringify(workExpList));
```

Comment:
There's no handling of potential errors that might arise, especially when dealing with localStorage (e.g., when the storage is full). Consider adding try-catch blocks around operations that might fail.